kratos/future/solving_strategies/builders/block_builder.h

```diff
@@ -0,0 +1,375 @@
+        const DofsArrayType& rEffectiveDofArray,
+        LinearSystemContainer<TSparseMatrixType, TSystemVectorType>& rLinearSystemContainer) override
+    {
+        // Calculate the effective LHS, RHS and solution vector
+        ApplyBlockBuildMasterSlaveConstraints(rLinearSystemContainer);
+
+        // Apply the Dirichlet BCs in a block way by leveraging the CSR matrix implementation
+        auto& r_eff_rhs = *(rLinearSystemContainer.pEffectiveRhs);
+        auto& r_eff_lhs = *(rLinearSystemContainer.pEffectiveLhs);
+        ApplyBlockBuildDirichletConditions(rEffectiveDofArray, r_eff_lhs, r_eff_rhs);
+    }
+
+    ///@}
+private:
+    ///@name Member Variables
+    ///@{
+
+    ScalingType mScalingType; // Diagonal scaling type
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    /**
+     * @brief Applies the master-slave constraints
+     * This method applies the master-slave constraints following a block-type build
+     * @param rLinearSystemContainer Auxiliary container with the linear system arrays
+     */
+    void ApplyBlockBuildMasterSlaveConstraints(LinearSystemContainer<TSparseMatrixType, TSystemVectorType> &rLinearSystemContainer)
+    {
+        const auto& r_model_part = this->GetModelPart();
+        const std::size_t n_constraints = r_model_part.NumberOfMasterSlaveConstraints();
+        if (n_constraints) { //FIXME: In here we should check the number of active constraints
+            // Get effective arrays
+            auto p_eff_dx = rLinearSystemContainer.pEffectiveDx;
+            auto p_eff_rhs = rLinearSystemContainer.pEffectiveRhs;
+
+            // Initialize the effective RHS
+            KRATOS_ERROR_IF(p_eff_rhs == nullptr) << "Effective RHS vector has not been initialized yet." << std::endl;
+            p_eff_rhs->SetValue(0.0);
+
+            // Initialize the effective solution vector
+            KRATOS_ERROR_IF(p_eff_dx == nullptr) << "Effective solution increment vector has not been initialized yet." << std::endl;
+            p_eff_dx->SetValue(0.0);
+
+            // Assign the master-slave relation matrix as the effective one since there are no other constraints
+            rLinearSystemContainer.pEffectiveT = rLinearSystemContainer.pConstraintsT;
+
+            // Apply constraints to RHS
+            auto p_rhs = rLinearSystemContainer.pRhs;
+            rLinearSystemContainer.pEffectiveT->TransposeSpMV(*p_rhs, *p_eff_rhs);
+
+            // Apply constraints to LHS
+            auto p_lhs = rLinearSystemContainer.pLhs;
+            auto p_LHS_T = AmgclCSRSpMMUtilities::SparseMultiply(*p_lhs, *rLinearSystemContainer.pEffectiveT);
+            auto p_transT = AmgclCSRConversionUtilities::Transpose(*rLinearSystemContainer.pEffectiveT);
+            rLinearSystemContainer.pEffectiveLhs = AmgclCSRSpMMUtilities::SparseMultiply(*p_transT, *p_LHS_T);
+
+            // // Compute the scale factor value
+            // //TODO: think on how to make this user-definable
```

Comment:
If not finished I will put an error here

Reply:
If we throw an error we cannot use it. In this regard, I note that this PR doesn't really need this method, but in the strategy we of course use it. I just wanted to  make the PRs as small as possible.